packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,494 @@
+
+Like the `IFRAME/RPC` or `POP/RPS`, our goal is to eventually get an `APPROVED` or `DECLINED` polling response, and technically this endpoint could return one of those immediately.
+
+But more than likely that isn't the case and it will be in a `PENDING` state (`PENDING` is not available to `IFRAME/RPC` or `POP/RPC`).
+When the polling response is `PENDING` it requires an `update` field that includes a service, `BackChannelRpc`, that FCL can use to request an updated `PollingResponse` from.
+FCL will use that `BackChannelRpc` to request a new `PollingResponse` which itself can be `APPROVED`, `DECLINED` or `PENDING`.
+If it is `APPROVED` or `DECLINED` FCL will halt and return/error, but if it is `PENDING` it will use the `BackChannelRpc` supplied in the new `PollingResponse` update field.
+It will repeat this cycle until it is either `APPROVED` or `DECLINED`.
+
+There is an additional feature that `HTTP/POST` enables in the first `PollingResponse` that is returned.
+This feature is the ability for FCL to render an iframe or popup, and it can be triggered by supplying a service `type: "VIEW/FRAME"` or `type: "VIEW/POP"` and the `endpoint` that the wallet wishes to render.
+
+### Polling Response
+
+```javascript
+// APPROVED
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "APPROVED",
+  data: ___, // what the service needs to send to FCL
+}
+
+// Declined
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "DECLINED",
+  reason: "Declined by user."
+}
+
+// Pending - Simple
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // will be included in the requests body
+    params: {}, // will be included in the requests url
+  }
+}
+
+// Pending - First Time with Local
+{
+  f_type: "PollingResponse",
+  f_vsn: "1.0.0",
+  status: "PENDING",
+  updates: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "back-channel-rpc",
+    endpoint: "https://____", // where post request will be sent
+    method: "HTTP/POST",
+    data: {},   // included in body of request
+    params: {}, // included as query params on endpoint
+  },
+  local: {
+    f_type: "Service",
+    f_vsn: "1.0.0",
+    type: "view/frame",
+    endpoint: "https://____", // the iframe that will be rendered,
+    data: {}, // sent to frame when ready
+    params: {}, // included as query params on endpoint
+  }
+}
+```
+
+### data and params
+
+`data` and `params` are information that the wallet can provide in the service config that FCL will pass back to the service.
+- `params` will be added onto the `endpoint` as query params.
+- `data` will be included in the body of the `HTTP/POST` request or in the `FCL:FRAME:READY:RESPONSE` for a `IFRAME/RPC` or `POP/RPC`
```

Comment:
Deprecation here as well. Should these reference the future direction `FCL:VIEW:READY:RESPONSE`